packages/devtools_app/lib/src/debugger/debugger_controller.dart

```diff
@@ -127,12 +138,56 @@ class DebuggerController extends DisposableController
   /// history).
   void _showScriptLocation(ScriptLocation scriptLocation) {
     _currentScriptRef.value = scriptLocation?.scriptRef;
+
+    parseCurrentScript();
+
     // We want to notify regardless of the previous scriptLocation, temporarily
     // set to null to ensure that happens.
     _scriptLocation.value = null;
     _scriptLocation.value = scriptLocation;
   }
 
+  Future<Script> getScriptForRef(ScriptRef ref) async {
+    final cachedScript = getScriptCached(ref);
+    if (cachedScript == null && ref != null) {
+      return await getScript(ref);
+    }
+    return cachedScript;
+  }
+
+  Future<void> parseCurrentScript() async {
```

Comment:
This looks like it can be private. We should also document what it's doing.

Reply:
renamed and added dart doc